### PR TITLE
Integrate @all

### LINKED ISSUE #13: Ability to @all
As a user, I would like to get everyone's attention to something of importance with `@everyone`

Not supported in GroupMe naively

Acceptance Criteria
-----------------
Channel with members P0, P1, P2 & P3
```
P1: @all Check this cool thing out!
Bot: @P0 @P2 @P3 
```
The bot's add should be true, notification causing @'s, not simply the text.

### PR DESCRIPTION
**Pull request by JosephMart**

### CODE REVIEW

File: src/services/GroupMeService.ts [hunk 1]
```diff
@@ -19,11 +19,12 @@ export class GroupMeService implements MessagingService {
    * Sends a message to GroupMe.
    * @param text
    */
-  async sendMessage(text: string): Promise<Either<Error, number>> {
+  async sendMessage(text: string, attachments: GroupMeAttachment[] = []): Promise<Either<Error, number>> {
```
Comment (tristan957) [inline src/services/GroupMeService.ts:54]: Any reason for default param over possibly undefined?

File: src/services/GroupMeService.ts [hunk 2]
```diff
@@ -39,6 +40,23 @@ export class GroupMeService implements MessagingService {
   }
 }
 
+export type GroupMeMention = { loci: [number, number]; type: "mentions"; user_ids: string[] };
```
Comment (tristan957) [inline src/services/GroupMeService.ts]: What does a mention attachment look like? Is loci an array of 2 values?

**Review by tristan957 — APPROVED** [review]

File: src/services/GroupMeService.ts (hunk 1 above)
Comment (JosephMart) [reply src/services/GroupMeService.ts:54]: https://github.com/JosephMart/loki-gm/pull/14/files#diff-0135c17db556ed19014403cc5c86b0a1R27

File: src/services/GroupMeService.ts (hunk 2 above)
Comment (JosephMart) [reply src/services/GroupMeService.ts]: yes https://github.com/JosephMart/loki-gm/issues/13#issuecomment-620257474

**Comment by tristan957** [comment]
Sounds good to me

File: src/services/GroupMeService.ts [hunk 3]
```diff
@@ -38,8 +40,76 @@ export class GroupMeService implements MessagingService {
...
+      return right(members);
+    } catch (e) {
+      return left(e as Error);
+    }
+  }
+
+  /**
+   * Get group info
+   */
+  async getGroupInfo(): Promise</*Either<Error, */ GroupMeGroupInfo> /*> */ {
```
Comment (tristan957) [inline src/services/GroupMeService.ts]: lel

File: src/services/GroupMeService.ts [hunk 4]
```diff
@@ -38,8 +40,76 @@ export class GroupMeService implements MessagingService {
...
+  share_qr_code_url: string;
+  members: GroupMeUser[];
+  messages: {
+    count: number;
+    last_message_id: string;
+    last_message_created_at: number;
+    preview: {
+      nickname: string;
+      text: string;
+      image_url: "https://i.groupme.com/852x852.jpeg.d82cfc5635944c2fb290d84486d99527";
```
Comment (tristan957) [inline src/services/GroupMeService.ts]: should this be a string type?

**Review by tristan957 — CHANGES_REQUESTED** [review]

File: src/services/GroupMeService.ts (hunk 4 above)
Comment (JosephMart) [reply src/services/GroupMeService.ts]: Yeah was sleepy 🥴

File: src/services/GroupMeService.ts (hunk 2 above)
Comment (JosephMart) [reply src/services/GroupMeService.ts]: it's actually [[int, int], [int, int]]. `loci.length === user_ids.length`

File: src/services/GroupMeService.ts (hunk 3 above)
Comment (JosephMart) [reply src/services/GroupMeService.ts]: Assigned to @tristan957 tofix :)

File: src/services/GroupMeService.ts [hunk 5]
```diff
@@ -37,8 +40,93 @@ export class GroupMeService implements MessagingService {
...
+  group_id: string;
+  name: string;
+  phone_number: string;
+  type: string;
+  description: string;
+  image_url: null;
+  creator_user_id: string;
+  created_at: number;
+  updated_at: number;
+  muted_until: null; // not sure about this
```
Comment (JosephMart) [inline src/services/GroupMeService.ts]: The response had `muted_until: null`, not sure if actual type is `string` or `number`

File: src/services/GroupMeService.ts [hunk 6]
```diff
@@ -37,8 +40,93 @@ export class GroupMeService implements MessagingService {
...
+      nickname: string;
+      text: string;
+      image_url: string;
+      attachments: GroupMeAttachment[];
+    };
+  };
+  max_members: number;
+};
+
+export type GroupMeMention = { loci: Array<[number, number]>; type: "mentions"; user_ids: string[] };
```
Comment (JosephMart) [inline src/services/GroupMeService.ts]: Is it good to "hardcode" string type like mentions here? Should to it to image and location below. Maybe they should be constants? Add helper constructor funcs?

Comment (tristan957) [reply src/services/GroupMeService.ts]: fine to me

File: src/services/GroupMeService.ts (hunk 5 above)
Comment (tristan957) [reply src/services/GroupMeService.ts]: Probably a nullable number

**Review by tristan957 — APPROVED** [review]

**Comment by tristan957** [comment]
Fixed the regex so all tests pass now. I would like to change how we are spitting the message back out in the groupme so that the position of @all is retained in the text whereas rn we remove it. I think there are too many edge cases that make it not worth.

```
Tristan Partin: @all hey guys we cool people
loki: @Tristan @Joseph: @all hey guys we cool people
```

**Comment by JosephMart** [comment]
I’m not sure how I feel about repeating the message anymore. The goal is just bring attention to the message with @all text and 99% of the time the message is going to be right above.

**Comment by tristan957** [comment]
Wish we could just send an up arroow emoji tbh, but I agree. Let's drop the message. Do you want to not tag the person who sent the @all message?
